cli/lib.rs

```diff
@@ -371,12 +371,17 @@ fn run_script(flags: DenoFlags, argv: Vec<String>) {
     js_check(worker.execute("denoMain()"));
     debug!("main_module {}", main_module);
 
+    let mut worker_ = worker.clone();
+
     worker
       .execute_mod_async(&main_module, false)
       .and_then(move |()| {
         js_check(worker.execute("window.dispatchEvent(new Event('load'))"));
-        worker.then(|result| {
+        worker.then(move |result| {
           js_check(result);
+          js_check(
+            worker_.execute("window.dispatchEvent(new Event('unload'))"),
```

Comment:
In the spec for onunload, so all workers get onunload event? The way this is implemented only the main worker will.

Reply:
That's correct, on MDN [Worker's](https://developer.mozilla.org/en-US/docs/Web/API/Worker) and MSN [GlobalEventHandlers](https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers) there's no information about `onunload` for workers.

CC @kitsonk

Reply:
Correct, even within the worker there is no sense of a page lifecycle, and it is an express design decision.  Workers are really designed to deal with `onmessage` and have that be a complete "lifecycle".  Service workers have a few more events that they directly support.  So in theory if you needed to shut down your workers, you would message them with something meaningful to clean up, so the only thing to do is ensure that all messages have been dispatched before exiting.